### DISCUSSION TITLE
Is there a way to configure usermode networking?

### DISCUSSION THREAD

**Discussion by Phx01** [virt-manager/virt-manager#738] [Q&A]
[QEMU's wiki](https://wiki.qemu.org/Documentation/Networking#User_Networking_.28SLIRP.29) states that the default SLIRP networking is using 10.0.2.0/24 with the first VM using 10.0.2.15.  Since this network conflicts with the physical network we have, I looked into ways on how to configure it to be on a different network.

According to their wiki, the related QEMU command line option would be:

    -netdev user,id=mynet0,net=192.168.76.0/24,dhcpstart=192.168.76.9

However, I fail to understand how I can convert that to libvirt XML/add it to virt-manager.  Is this possible?  If so, what would the XML snippet be to implement it?

The intention is to build a simple private VM on a shared server, but I could not wrap my head around securing an active system VM from prying eyes, e.g. people could simply use VNC or similar to connect to my active session, taking it over simply after logging on to the server with their account.

PS: If there is a guide I can follow to secure a system VM, it would be a good alternative.

**Comment by RaymiiOrg** [comment]
Here is some more information:

https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/6/html/virtualization_administration_guide/sub-sub-section-libvirt-dom-xml-devices-network-interfaces-userspace-slirp-stack#sub-sub-section-libvirt-dom-xml-devices-Network-interfaces-userspace-slirp-stack

```
 The user-space SLIP stack parameters are defined in the following part of the domain XML::


  ...
  <devices>
    <interface type='user'/>
    ...
    <interface type='user'>
      <mac address="00:11:22:33:44:55"/>
    </interface>
  </devices>
  ...
```
